### PR TITLE
fix: remove implicit state toggling for spatial underlay in favor of explicit state setting

### PR DESCRIPTION
**Pull request by seve**
Old redux action handling image underlay display state was error-prone. This PR removes the reducer's implicit state toggling with explicit state sent via the action payload

### CODE REVIEW

**Review by kaloster — APPROVED** [review]
Looks good, thanks!

File: client/src/reducers/controls.ts [hunk 1]
```diff
@@ -457,7 +457,7 @@ const Controls = (
     case "toggle image underlay": {
       return {
         ...state,
-        imageUnderlay: !state.imageUnderlay,
```
Comment (tihuan) [inline client/src/reducers/controls.ts:460]: let's check the existing state and only return a new state if it's different, so we don't unnecessarily trigger a re-render.

E.g.,

```ts
case "toggle image underlay": {
  if (state.imageUnderlay === action.toggle) return state;

  return { ... }
```

**Review by tihuan — COMMENTED** [review]
Looks great! Just one optimization to do 👌 Thanks so much, Seve! 🚀

**Review by tihuan — APPROVED** [review]
Thanks for the update!!

**Pull request merged**